identity-hub-client/src/test/java/org/eclipse/dataspaceconnector/identityhub/client/IdentityHubClientImplIntegrationTest.java

```diff
@@ -39,24 +43,27 @@ public class IdentityHubClientImplIntegrationTest {
     @BeforeEach
     void setUp() {
         var okHttpClient = TestUtils.testOkHttpClient();
-        client = new IdentityHubClientImpl(okHttpClient, OBJECT_MAPPER);
+        client = new IdentityHubClientImpl(okHttpClient, OBJECT_MAPPER, new ConsoleMonitor());
     }
 
     @Test
     void addAndQueryVerifiableCredentials() throws Exception {
         var credential = VerifiableCredential.Builder.newInstance().id(VERIFIABLE_CREDENTIAL_ID).build();
-        addVerifiableCredential(credential);
-        getVerifiableCredential(credential);
+        SignedJWT jws = buildSignedJwt(credential, FAKER.internet().url(), FAKER.internet().url());
+
+        addVerifiableCredential(jws);
+
+        getVerifiableCredential(jws);
     }
 
-    private void addVerifiableCredential(VerifiableCredential credential) throws IOException {
-        var statusResult = client.addVerifiableCredential(API_URL, credential);
+    private void addVerifiableCredential(SignedJWT jws) throws IOException {
+        var statusResult = client.addVerifiableCredential(API_URL, jws);
         assertThat(statusResult.succeeded()).isTrue();
     }
 
-    private void getVerifiableCredential(VerifiableCredential credential) throws IOException {
+    private void getVerifiableCredential(SignedJWT jws) throws IOException {
         var statusResult = client.getVerifiableCredentials(API_URL);
         assertThat(statusResult.succeeded()).isTrue();
-        assertThat(statusResult.getContent()).usingRecursiveFieldByFieldElementComparator().contains(credential);
+        assertThat(statusResult.getContent().stream().map(JWSObject::serialize)).contains(jws.serialize());
```

Comment:
```suggestion
        assertThat(statusResult.getContent().stream().usingRecursiveComparison().contains(jws);
```
or is there a particular reason to serialise? The proposed form is stabler and more natural

Reply:
Agree and changed